sensor/common/processindicator/component.go

```diff
@@ -0,0 +1,165 @@
+package processindicator
+
+import (
+	"io"
+	"strings"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/stringutils"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+type componentImpl struct {
+	common.SensorComponent
+
+	processPipeline *Pipeline
+	signalMessages  <-chan *storage.ProcessSignal
+	processMessages <-chan *sensor.ProcessSignal
+	writer          io.Writer
+
+	stopper concurrency.Stopper
+}
+
+type Option func(*componentImpl)
+
+// WithTraceWriter sets a trace writer that will write the messages received from collector.
+func WithTraceWriter(writer io.Writer) Option {
+	return func(cmp *componentImpl) {
+		cmp.writer = writer
+	}
+}
+
+func New(pipeline *Pipeline, signalMessages <-chan *storage.ProcessSignal, processMessages <-chan *sensor.ProcessSignal, opts ...Option) common.SensorComponent {
+	cmp := &componentImpl{
+		processPipeline: pipeline,
+		signalMessages:  signalMessages,
+		processMessages: processMessages,
+		writer:          nil,
+		stopper:         concurrency.NewStopper(),
+	}
+	for _, o := range opts {
+		o(cmp)
+	}
+	return cmp
+}
+
+func (c *componentImpl) Start() error {
+	go c.run()
+	return nil
+}
+
+func (c *componentImpl) Stop(_ error) {
+	c.processPipeline.Shutdown()
+	if !c.stopper.Client().Stopped().IsDone() {
+		defer func() {
+			_ = c.stopper.Client().Stopped().Wait()
+		}()
+	}
+	c.stopper.Client().Stop()
+}
+
+func (c *componentImpl) Notify(e common.SensorComponentEvent) {
+	log.Info(common.LogSensorComponentEvent(e, "processindicator"))
+	c.processPipeline.Notify(e)
+}
+
+func (c *componentImpl) Capabilities() []centralsensor.SensorCapability {
+	return nil
+}
+func (c *componentImpl) ProcessMessage(_ *central.MsgToSensor) error {
+	return nil
+}
+
+func (c *componentImpl) ResponsesC() <-chan *message.ExpiringMessage {
+	return c.processPipeline.indicators
+}
+
+func (c *componentImpl) run() {
+	defer c.stopper.Flow().ReportStopped()
+
+	for {
+		select {
+		case msg := <-c.processMessages:
+			c.processMsg(sensorIntoStorageSignal(msg))
+		case msg := <-c.signalMessages:
+			c.processMsg(msg)
```

Comment:
Normally in these buffers we track when a message is added to the queue and when is removed. That way we can see if the queue is growing and how big it is (added vs removed messages). I think we should increase the `totalProcessesCounter` with the operation `remove` to track how many messages were processed.